homeassistant/components/folder_watcher/__init__.py

```diff
@@ -115,20 +142,29 @@ def on_closed(self, event):
 class Watcher:
     """Class for starting Watchdog."""
 
-    def __init__(self, path, patterns, hass):
+    def __init__(self, path: str, patterns: list[str], hass: HomeAssistant) -> None:
         """Initialise the watchdog observer."""
+        self.hass = hass
         self._observer = Observer()
         self._observer.schedule(
             create_event_handler(patterns, hass), path, recursive=True
         )
-        hass.bus.listen_once(EVENT_HOMEASSISTANT_START, self.startup)
-        hass.bus.listen_once(EVENT_HOMEASSISTANT_STOP, self.shutdown)
+        hass.bus.listen_once(EVENT_HOMEASSISTANT_START, self.async_startup)
+        hass.bus.listen_once(EVENT_HOMEASSISTANT_STOP, self.async_shutdown)
```

Comment:
When trying to set up the integration via config flow, I get this exception:

```
ERROR (MainThread) [homeassistant.config_entries] Error setting up entry Folder Watcher /Downloads for folder_watcher
Traceback (most recent call last):
  File "<REPODIR>/homeassistant/config_entries.py", line 406, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<REPODIR>/homeassistant/components/folder_watcher/__init__.py", line 78, in async_setup_entry
    Watcher(path, patterns, hass)
  File "<REPODIR>/homeassistant/components/folder_watcher/__init__.py", line 152, in __init__
    hass.bus.listen_once(EVENT_HOMEASSISTANT_START, self.async_startup)
  File "<REPODIR>/homeassistant/core.py", line 1261, in listen_once
    async_remove_listener = run_callback_threadsafe(
                            ^^^^^^^^^^^^^^^^^^^^^^^^
  File "<REPODIR>/homeassistant/util/async_.py", line 41, in run_callback_threadsafe
    raise RuntimeError("Cannot be called from within the event loop")
RuntimeError: Cannot be called from within the event loop
```

(But maybe it is just in my environment, I don't know)

Reply:
Appreciate you testing! Please do again if possible after my changes 👍